modules/ml/test/test_rtrees.cpp

```diff
@@ -95,6 +95,24 @@ TEST(ML_RTrees, 11142_sample_weights_classification)
     EXPECT_GE(error_with_weights, error_without_weights);
 }
 
+TEST(ML_RTrees, bug_12974_throw_exception_when_predict_different_feature_count)
+{
+    // create a simple 2 feature dataset and train + save the model
+    cv::Ptr<RTrees> model = RTrees::create();
+    Mat samples(2, 2, CV_32FC1);
+    samples.at<float>(0,0) = 0;
+    samples.at<float>(0,1) = 0;
+    samples.at<float>(1,0) = 1;
+    samples.at<float>(1,1) = 1;
+    Mat responses(2, 1, CV_32S);
+    responses.at<float>(0) = 0;
+    responses.at<float>(1) = 1;
+    cv::Ptr<TrainData> trainData = TrainData::create(samples, cv::ml::ROW_SAMPLE, responses);
+    model->train(trainData);
+    // load the model and try to predict on a 1 feature sample
+    Mat test(1, 1, CV_32FC1);
```

Comment:
Does it make sense to check multiple sizes in a loop (`1..n-1` and `n+1`)?

Reply:
@mshabunin not sure I understand what you mean?

Reply:
Ah! You mean assert that the test fails for < n and > n. Good idea.

Reply:
Done - I increased the number of features to make it more comprehensive too (now we have 5 in training, and test on 0-4 and 6).